rasa_core/events/__init__.py

```diff
@@ -796,3 +831,49 @@ def _from_parameters(cls, parameters):
         except KeyError as e:
             raise ValueError("Failed to parse agent uttered event. "
                              "{}".format(e))
+
+
+class Form(Event):
+    type_name = "form"
+
+    def __init__(self, form_name, timestamp=None):
+        super(Form, self).__init__(timestamp)
+        self.form = form_name
+        self.postponed = True
+
+    def postponed_apply_to(self, tracker):
+        # type: (DialogueStateTracker) -> None
+        tracker.form(self.form)
+
+    def as_story_string(self):
+        props = json.dumps({"form_name": self.form})
+        return "{name}{props}".format(name=self.type_name, props=props)
+
+    @classmethod
+    def _from_story_string(cls, parameters):
+        """Called to convert a parsed story line into an event."""
+        return [Form(parameters.get("form_name"),
+                              parameters.get("timestamp"))]
+
+    def as_dict(self):
+        d = super(Form, self).as_dict()
+        d.update({"form_name": self.form})
+        return d
+
+
+class ActionExecutionFailed(Event):
+    type_name = 'action_execution_failed'
+
+    def __init__(self, action_name, timestamp=None):
```

Comment:
this should have the same signature as the action_executed (so adding policy and confidence)